Jenkinsfile

```diff
@@ -45,7 +45,7 @@ config = [
             tags: ['docker'],
             builds: ['release'],
             extraBuildFlags: [
-                'CMAKE_CXX_FLAGS:STRING=-Werror -Wno-maybe-uninitialized -Wno-array-bounds',
+                'CMAKE_CXX_FLAGS:STRING=-Werror -Wno-maybe-uninitialized -Wno-array-bounds -Wno-free-nonheap-object',
```

Comment:
Did we have a false positive here?

Reply:
This false positive surfaced after moving the `lp.frame` test to the new framework:

```
[ 78%] Building CXX object libcaf_net/CMakeFiles/caf-net-test.dir/caf/net/lp/frame.test.cpp.o
In file included from /usr/include/c++/12.2.1/x86_64-alpine-linux-musl/bits/c++allocator.h:33,
                 from /usr/include/c++/12.2.1/bits/allocator.h:46,
                 from /usr/include/c++/12.2.1/memory:64,
                 from /users/jenkins/workspace/work_topic_neverlord_net-testing@2/sources/libcaf_core/caf/fwd.hpp:11,
                 from /users/jenkins/workspace/work_topic_neverlord_net-testing@2/sources/libcaf_core/caf/async/fwd.hpp:7,
                 from /users/jenkins/workspace/work_topic_neverlord_net-testing@2/sources/libcaf_core/caf/chunk.hpp:7,
                 from /users/jenkins/workspace/work_topic_neverlord_net-testing@2/sources/libcaf_net/caf/net/lp/frame.hpp:7,
                 from /users/jenkins/workspace/work_topic_neverlord_net-testing@2/sources/libcaf_net/caf/net/lp/frame.test.cpp:5:
In member function 'void std::__new_allocator<_Tp>::deallocate(_Tp*, size_type) [with _Tp = std::byte]',
    inlined from 'static void std::allocator_traits<std::allocator<_Tp1> >::deallocate(allocator_type&, pointer, size_type) [with _Tp = std::byte]' at /usr/include/c++/12.2.1/bits/alloc_traits.h:496:23,
    inlined from 'void std::_Vector_base<_Tp, _Alloc>::_M_deallocate(pointer, std::size_t) [with _Tp = std::byte; _Alloc = std::allocator<std::byte>]' at /usr/include/c++/12.2.1/bits/stl_vector.h:387:19,
    inlined from 'std::_Vector_base<_Tp, _Alloc>::~_Vector_base() [with _Tp = std::byte; _Alloc = std::allocator<std::byte>]' at /usr/include/c++/12.2.1/bits/stl_vector.h:366:15,
    inlined from 'std::vector<_Tp, _Alloc>::~vector() [with _Tp = std::byte; _Alloc = std::allocator<std::byte>]' at /usr/include/c++/12.2.1/bits/stl_vector.h:733:7,
    inlined from 'virtual void {anonymous}::test_31::do_run()' at /users/jenkins/workspace/work_topic_neverlord_net-testing@2/sources/libcaf_net/caf/net/lp/frame.test.cpp:39:18:
/usr/include/c++/12.2.1/bits/new_allocator.h:158:33: error: 'void operator delete(void*, std::size_t)' called on pointer '<unknown>' with nonzero offset [1, 9223372036854775807] [-Werror=free-nonheap-object]
  158 |         _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n));
      |                                 ^
In member function '_Tp* std::__new_allocator<_Tp>::allocate(size_type, const void*) [with _Tp = std::byte]',
    inlined from 'static _Tp* std::allocator_traits<std::allocator<_Tp1> >::allocate(allocator_type&, size_type) [with _Tp = std::byte]' at /usr/include/c++/12.2.1/bits/alloc_traits.h:464:28,
    inlined from 'std::_Vector_base<_Tp, _Alloc>::pointer std::_Vector_base<_Tp, _Alloc>::_M_allocate(std::size_t) [with _Tp = std::byte; _Alloc = std::allocator<std::byte>]' at /usr/include/c++/12.2.1/bits/stl_vector.h:378:33,
    inlined from 'void std::vector<_Tp, _Alloc>::_M_range_initialize(_ForwardIterator, _ForwardIterator, std::forward_iterator_tag) [with _ForwardIterator = const std::byte*; _Tp = std::byte; _Alloc = std::allocator<std::byte>]' at /usr/include/c++/12.2.1/bits/stl_vector.h:1687:25,
    inlined from 'std::vector<_Tp, _Alloc>::vector(_InputIterator, _InputIterator, const allocator_type&) [with _InputIterator = const std::byte*; <template-parameter-2-2> = void; _Tp = std::byte; _Alloc = std::allocator<std::byte>]' at /usr/include/c++/12.2.1/bits/stl_vector.h:706:23,
    inlined from 'std::vector<std::byte> {anonymous}::to_vec(caf::span<const T>) [with T = std::byte]' at /users/jenkins/workspace/work_topic_neverlord_net-testing@2/sources/libcaf_net/caf/net/lp/frame.test.cpp:20:61,
    inlined from 'virtual void {anonymous}::test_31::do_run()' at /users/jenkins/workspace/work_topic_neverlord_net-testing@2/sources/libcaf_net/caf/net/lp/frame.test.cpp:39:18:
/usr/include/c++/12.2.1/bits/new_allocator.h:137:55: note: returned from 'void* operator new(std::size_t)'
  137 |         return static_cast<_Tp*>(_GLIBCXX_OPERATOR_NEW(__n * sizeof(_Tp)));
      |                                                       ^
cc1plus: all warnings being treated as errors
```

The line in question just calls `resize` on a vector. No idea what GCC is confused about here. Hence, decided to ignore this as noise.